framework/src/main/java/org/checkerframework/framework/util/JavaExpressionParseUtil.java

```diff
@@ -348,22 +392,22 @@ public JavaExpression visit(ArrayAccessExpr expr, JavaExpressionContext context)
             }
             TypeMirror componentType = ((ArrayType) arrayType).getComponentType();
 
-            JavaExpression index = expr.getIndex().accept(this, context);
+            JavaExpression index = expr.getIndex().accept(this, null);
```

Comment:
The code above throws an exception if the "a" part of "a[i]" is not an array.  Should it also check that the "i" part is an integer?

(There are probably other places that the type could also be checked.  I think that wrong types won't cause crashes, but they may be confusing to users, which would be a reason to warn during parsing, when it is easy to warn.)

Reply:
Let's defer this, I opened an issue for it https://github.com/typetools/checker-framework/issues/4400.